domain/domain.go

```diff
@@ -453,6 +468,28 @@ func (do *Domain) mustRestartSyncer() error {
 	}
 }
 
+// mustReload tries to Reload the schema, it returns until it's successful or the domain is closed.
+// it returns nil when it is sucessful, returns err when the domain is closed.
+func (do *Domain) mustReload() error {
+	for {
+		err := do.Reload()
+		if err == nil {
+			log.Infof("[ddl] mustReload succeed.")
+			return nil
+		}
+
+		// If the domain is closed, we returns immediately.
+		select {
+		case <-do.exit:
+			return errors.Trace(err)
+		default:
+		}
+
+		time.Sleep(time.Second)
```

Comment:
So once we lost connection to etcd, the service unavailable time is at lease several seconds ?

Reply:
Why do you think so? The service restore depends on when TiDB can reconnection to etcd.

Reply:
Could this retry duration be smaller?

Reply:
What is the proper retry duration do you think?